src/grt/src/MakeWireParasitics.cpp

```diff
@@ -491,47 +491,31 @@ float MakeWireParasitics::getNetSlack(odb::dbNet* net)
 std::vector<int> MakeWireParasitics::routeLayerLengths(odb::dbNet* db_net) const
 {
   NetRouteMap& routes = grouter_->getRoutes();
-  std::vector<int> layer_lengths(grouter_->getMaxRoutingLayer() + 1);
+
+  // dbu wirelength for wires, via count for vias
+  std::vector<int> layer_lengths(tech_->getLayerCount());
+
   if (!db_net->getSigType().isSupply()) {
     GRoute& route = routes[db_net];
     std::set<RoutePt> route_pts;
     for (GSegment& segment : route) {
       if (segment.isVia()) {
-        route_pts.insert(
-            RoutePt(segment.init_x, segment.init_y, segment.init_layer));
-        route_pts.insert(
-            RoutePt(segment.final_x, segment.final_y, segment.final_layer));
+        auto& s = segment;
+        // Mimic makeRouteParasitics
+        int min_layer = min(s.init_layer, s.final_layer);
+        odb::dbTechLayer* cut_layer
+            = tech_->findRoutingLayer(min_layer)->getUpperLayer();
+        layer_lengths[cut_layer->getNumber()] += 1;
```

Comment:
Is it worthwhile to track the cut layers given our discussion of via resistance?